src/Illuminate/Foundation/Application.php

```diff
@@ -520,6 +520,16 @@ public function runningInConsole()
         return php_sapi_name() === 'cli' || php_sapi_name() === 'phpdbg';
     }
 
+    /**
+     * Determine if the application is running in the console as http server.
+     *
+     * @return bool
+     */
+    public function runningInConsoleAsHttpServer()
```

Comment:
It seems like we do not need this new method here.

You could just add the condition to `runningInConsole()`

Reply:
I think we should not change methods' behavior like `runningInConsole` , it's used everywhere in some projects(my projects), add an extra condition may cause unexpected problems.
Am I too much cautious?

Reply:
It looks like your condition is just another special case for `runningInConsole` that we were not aware before.

Reply:
2 reasons:
1. Compatibility
2. Script running in `cli-server` sapi is `running in console` indeed, it's weird if gets `false` when calling this method in `cli-server`, so add a new method `runningInConsoleAsHttpServer`  is reasonable.

Reply:
I am not following you?
What do you mean by `compatibility`? Can you explain which compatibility problems we could find if you add the extra case into the existing method?

And if the script is indeed running in a console then it should not change behaviour after adding this new case.

Reply:
This still doenst make sense to me. If the HTTP request is received - it is NOT "running in console" (because it came via HTTP).

The underlying server running is not related to the HTTP request.